### PR TITLE
Add gcs index support

### LINKED ISSUE #300: Add support for more index hosting backends (e.g., GCS, S3, etc.)
**What would you like to be added**:  

Support for retrieving index files from storage backends using protocols other than plain HTTP/S. For example GCS or S3 storage for cloud native environments. Additionally support for authentication mechanisms for HTTP/S backends would also be helpful for private rule sets/plugins hosted via HTTP/S.

**Why is this needed**:

Currently index files can only be retrieved via HTTP/S (no authentication). This is useful for public rule sets, but private rule sets would require both some sort of authentication support and ideally support for different more cloud native storage backends.

### PR DESCRIPTION
**Pull request by max-frank**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind documentation

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area library

/area cli

**What this PR does / why we need it**:

This PR adds support for using Google Cloud Storage as backend for storing *falcoctl* index files.

**Which issue(s) this PR fixes**:

Fixes #300

**Special notes for your reviewer**:

Note I had to move some of the pre-existing logic into their own sub-packages to avoid import loops.
Specifically the `index.fetch` and `index.index` packages are just from moving stuff around.

### CODE REVIEW

**Comment by max-frank** [comment]
@MickaelFontes this might be useful for your use case as well maybe.

**Comment by alacuku** [comment]
Hi @max-frank, could you please rebase on top of the main branch?

**Comment by max-frank** [comment]
> Hi @max-frank, could you please rebase on top of the main branch?

@alacuku done 👍

File: pkg/index/fetch/gcs/structs.go
Comment (alacuku) [inline pkg/index/fetch/gcs/structs.go:1]: Could we come up with a more meaningful name for this file? Something like `types.go`?

**Review by alacuku — COMMENTED** [review]
@MickaelFontes, awesome work!

Left a minor comment.

File: pkg/index/fetch/gcs/structs.go
Comment (max-frank) [reply pkg/index/fetch/gcs/structs.go:1]: Done

**Review by alacuku — APPROVED** [review]
/LGTM

**Review by FedeDP — APPROVED** [review]
LGTM!
Great work, and thanks @alacuku for the review :)
/approve

**Pull request merged**